Fix overflow when texture size is smaller than block size

eranzhao opened the pull request:
If the texture size is smaller than the block size, it can cause an overflow. For example, a texture with size of 4x4 but format is  ASTC 6x6.

<!--
Before submitting a pull request you are strongly recommended to read the
docs/CONTRIBUTING.md file which gives some information on how to prepare a
change:

https://github.com/baldurk/renderdoc/blob/v1.x/docs/CONTRIBUTING.md

For small changes you don't have to read the document end to end, but should at
least look at the sections on how to ensure your code and commits are formatted
according to the style requirements.
-->

cmannett85-arm commented:
Hi @eranzhao, the ASTC spec requires that the data must be an integer of the block size, so even if the image dimensions are smaller than the block, the underlying data must be block aligned.  What prompted this PR?  Are you seeing a particular error?

baldurk commented:
Please also stop opening and closing PRs with no notice. There is nothing that requires re-creating a PR, if you need to fix any commits on the branch you can do so by force pushing.

eranzhao commented:
是的，我们在使用UE4的时候，贴图可以指定ASTC 6x6的格式，然后部分贴图的尺寸是4x4的，这个在renderdoc启动抓帧的时候，产生了溢出。造成的后果就是写坏了内存，在后续使用内存的时候，造成了崩溃。

Yes, when using UE4, we specified the ASTC 6x6 format for textures, but some textures were sized at 4x4. This caused an overflow when capturing frames with RenderDoc, resulting in memory corruption. Consequently, this led to crashes during subsequent memory usage.

以下是错误的信息：

The following is the wrong message:

![image](https://github.com/user-attachments/assets/141022d7-513d-40f0-b5eb-528c764b1c8d)
![image](https://github.com/user-attachments/assets/c25c0e02-cfc2-4d8f-8fe4-e5f5bd1ef9c3)

但是这个错误是可以忽略的，原因在前面。

But this error can be ignored for the reasons mentioned above.

我尝试修改源码定位这个问题，输出一些数据：

I tried to modify the source code to locate this problem and output some data:

![27d68355461bf0ddf3a775949a3310b](https://github.com/user-attachments/assets/e04a04c6-ea58-4f00-a548-ad8176dab03d)

我很认同你所说的数据必须对齐的观点，但是我目前还不能完全确定产生这个问题的根本原因。我关闭了这个PR是因为，虽然我通过了前段的问题，但是仍然有新的问题产生。当我搞清楚其他的问题的时候，我希望提交一个新的PR，来修复这个问题。

I agree with your point that the data must be aligned, but I'm not entirely sure yet what the root cause of this problem is. I closed this PR because, although I passed the previous questions, new questions still arise. When I figure out the other issues, I hope to submit a new PR to fix this issue.

eranzhao commented:
> Please also stop opening and closing PRs with no notice. There is nothing that requires re-creating a PR, if you need to fix any commits on the branch you can do so by force pushing.

抱歉造成了困扰，我将再努力下。

Sorry for the trouble, I will try harder.

eranzhao committed:
Fix overflow when texture size is smaller than block size

If the texture size is smaller than the block size, it can cause an overflow. For example, a texture size of 4x4 with a format of ASTC 6x6.
